src/components/ModalWithForm.js

```diff
@@ -6,6 +6,10 @@ export default class ModalWithForm extends Modal {
     this._modalForm = this._modalElement.querySelector(".modal__form");
     this._handleFormSubmit = handleFormSubmit;
   }
+  open() {
+    this._modalForm.querySelector(".modal__button").innerText = "Save";
```

Comment:
* You need to return the **default** button text only in blocks `finally` so that the button could get the text back in any case: `success` or `failure`.